compiler-plugin/src/main/java/io/ballerina/stdlib/http/compiler/codeaction/AddAnnotatedParam.java

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright (c) 2022, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.ballerina.stdlib.http.compiler.codeaction;
+
+import io.ballerina.compiler.syntax.tree.FunctionDefinitionNode;
+import io.ballerina.compiler.syntax.tree.FunctionSignatureNode;
+import io.ballerina.compiler.syntax.tree.NonTerminalNode;
+import io.ballerina.compiler.syntax.tree.SyntaxTree;
+import io.ballerina.projects.plugins.codeaction.CodeAction;
+import io.ballerina.projects.plugins.codeaction.CodeActionArgument;
+import io.ballerina.projects.plugins.codeaction.CodeActionContext;
+import io.ballerina.projects.plugins.codeaction.CodeActionExecutionContext;
+import io.ballerina.projects.plugins.codeaction.CodeActionInfo;
+import io.ballerina.projects.plugins.codeaction.DocumentEdit;
+import io.ballerina.stdlib.http.compiler.Constants;
+import io.ballerina.tools.diagnostics.Diagnostic;
+import io.ballerina.tools.text.LinePosition;
+import io.ballerina.tools.text.LineRange;
+import io.ballerina.tools.text.TextDocument;
+import io.ballerina.tools.text.TextDocumentChange;
+import io.ballerina.tools.text.TextEdit;
+import io.ballerina.tools.text.TextRange;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+
+import static io.ballerina.compiler.syntax.tree.SyntaxKind.FUNCTION_SIGNATURE;
+import static io.ballerina.compiler.syntax.tree.SyntaxKind.RESOURCE_ACCESSOR_DEFINITION;
+
+/**
+ * Abstract implementation of code action to add an annotated parameter to resource signature.
+ */
+public abstract class AddAnnotatedParam implements CodeAction {
```

Comment:
The class name should be a noun. Apparently previously added classes also starts with a verb. Is that the common practice for CodeAction classes?

Reply:
In [language-server code actions](https://github.com/ballerina-platform/ballerina-lang/tree/master/language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/codeaction/providers), the class names starts like a verb and end with `CodeAction` (For example `AddAnnotatedParameterCodeAction`)
Shall we follow the same convention?

Reply:
+1

Reply:
Fixed in [0c4163a](https://github.com/ballerina-platform/module-ballerina-http/pull/880/commits/0c4163a781d4b39592292f3c4b0fbf10b2b469f7)